### ISSUE TITLE
Command 'mediawiki-reload' (M-g) fails with 'readapidenied' after error "Bad token" (#7) occurs.

### ISSUE THREAD

**Issue by DrChr** [hexmode/mediawiki-el#8]
This is issue was noted while trying to use a proposed workaround for issue #7.

Situation:
- As from #7: Editing a wiki page in Emacs, saving fails after a while ("Bad token")
- Attempting the workaround 'mediawiki-reload' (M-g) to reload the buffer

The problem is that the reload fails with the error 'readapidenied'. I've attached the content of the  debug buffer at the end, here's the relevant part from buffer _Messages_:

```
Posting to: https://xxx.yyy.com/wiki/api.php
Contacting host: xxx.yyy.com:443
Examine the ' *http xxx.yyy.com:443*-986828' buffer called from 'url-http-response-post-process:3'.
mediawiki-api-call: The server encountered an error: (readapidenied) You need read permission to use this module
```

If I at this point try to use 'mediawiki-site' to connect to my wiki, it fails with 'readapidenied'.
However, if I connect to a _different_ wiki, e.g. wikipedia, and _then_ again connects to my wiki it works.

Any suggestion on how to troubleshoot this further?

Regards,
Christian

Below the content of the debug buffer:

```
HTTP/1.1 200 OK
Date: Mon, 18 Jan 2016 10:08:01 GMT
Server: Apache
X-Content-Type-Options: nosniff
MediaWiki-API-Error: readapidenied
X-Frame-Options: DENY
Cache-Control: private
Vary: Accept-Encoding
Content-Encoding: gzip
X-Powered-By: PleskLin
Content-Length: 114
Connection: close
Content-Type: text/xml; charset=utf-8

<?xml version="1.0"?><api><error code="readapidenied" info="You need read permission to use this module" /></api>
```

**Comment by hexmode** [comment]
Thanks for the additional details.
Could you tell me what version of MediaWiki the site you are trying to edit runs?

Sent from my Sprint Samsung Galaxy S® 6.-------- Original message --------From: DrChr <notifications@github.com> Date: 1/18/2016  4:17 AM  (GMT-06:00) To: hexmode/mediawiki-el <mediawiki-el@noreply.github.com> Subject: [mediawiki-el] Command 'mediawiki-reload' (M-g) fails with
  'readapidenied' after error "Bad token" (#7) occurs. (#8) 
This is issue was noted while trying to use a proposed workaround for issue #7

Situation:


As from #7: Editing a wiki page in Emacs, saving fails after a while ("Bad token")
Attempting the workaround 'mediawiki-reload' (M-g) to reload the buffer


The problem is that the reload fails with the error 'readapidenied' I've attached the content of the  debug buffer at the end, here's the relevant part from buffer Messages:

Posting to: https://xxxyyycom/wiki/apiphp
Contacting host: xxxyyycom:443
Examine the ' *http xxxyyycom:443*-986828' buffer called from 'url-http-response-post-process:3'
mediawiki-api-call: The server encountered an error: (readapidenied) You need read permission to use this module


If I at this point try to use 'mediawiki-site' to connect to my wiki, it fails with 'readapidenied'

However, if I connect to a different wiki, eg wikipedia, and then again connects to my wiki it works

Any suggestion on how to troubleshoot this further?

Regards,

Christian

Below the content of the debug buffer:

HTTP/11 200 OK
Date: Mon, 18 Jan 2016 10:08:01 GMT
Server: Apache
X-Content-Type-Options: nosniff
MediaWiki-API-Error: readapidenied
X-Frame-Options: DENY
Cache-Control: private
Vary: Accept-Encoding
Content-Encoding: gzip
X-Powered-By: PleskLin
Content-Length: 114
Connection: close
Content-Type: text/xml; charset=utf-8

<?xml version="10"?><api><error code="readapidenied" info="You need read permission to use this module" /></api>

**Comment by DrChr** [comment]
From the page Special:Version I get the following:

Product Version MediaWiki <https://www.mediawiki.org/> 1.23.6 PHP
<http://www.php.net/> 5.5.9-1ubuntu4.9 (cgi-fcgi) MySQL
<http://www.mysql.com/> 5.5.43-0ubuntu0.14.04.1


On 18 January 2016 at 16:44, Mark A. Hershberger <notifications@github.com>
wrote:

>
> Thanks for the additional details.
> Could you tell me what version of MediaWiki the site you are trying to
> edit runs?
>
> Sent from my Sprint Samsung Galaxy S® 6.-------- Original message
> --------From: DrChr <notifications@github.com> Date: 1/18/2016 4:17 AM
> (GMT-06:00) To: hexmode/mediawiki-el <mediawiki-el@noreply.github.com>
> Subject: [mediawiki-el] Command 'mediawiki-reload' (M-g) fails with
> 'readapidenied' after error "Bad token" (#7) occurs. (#8)
> This is issue was noted while trying to use a proposed workaround for
> issue #7
>
> Situation:
>
>
> As from #7: Editing a wiki page in Emacs, saving fails after a while ("Bad
> token")
> Attempting the workaround 'mediawiki-reload' (M-g) to reload the buffer
>
>
> The problem is that the reload fails with the error 'readapidenied' I've
> attached the content of the debug buffer at the end, here's the relevant
> part from buffer Messages:
>
> Posting to: https://xxxyyycom/wiki/apiphp
> Contacting host: xxxyyycom:443
> Examine the ' *http xxxyyycom:443*-986828' buffer called from
> 'url-http-response-post-process:3'
> mediawiki-api-call: The server encountered an error: (readapidenied) You
> need read permission to use this module
>
>
> If I at this point try to use 'mediawiki-site' to connect to my wiki, it
> fails with 'readapidenied'
>
> However, if I connect to a different wiki, eg wikipedia, and then again
> connects to my wiki it works
>
> Any suggestion on how to troubleshoot this further?
>
> Regards,
>
> Christian
>
> Below the content of the debug buffer:
>
> HTTP/11 200 OK
> Date: Mon, 18 Jan 2016 10:08:01 GMT
> Server: Apache
> X-Content-Type-Options: nosniff
> MediaWiki-API-Error: readapidenied
> X-Frame-Options: DENY
> Cache-Control: private
> Vary: Accept-Encoding
> Content-Encoding: gzip
> X-Powered-By: PleskLin
> Content-Length: 114
> Connection: close
> Content-Type: text/xml; charset=utf-8
>
> <?xml version="10"?><api><error code="readapidenied" info="You need read
> permission to use this module" /></api>
>
>
> —
> Reply to this email directly or view it on GitHub.
>
>
>
>
>
>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hexmode/mediawiki-el/issues/8#issuecomment-172565630>.
>



-- 
Christian Ridderström,     +46-70 687 39 44

**Comment by hexmode** [comment]
Are you still seeing this error? Have you upgraded to 1.27?

**Comment by hexmode** [comment]
Think this is fixed by the same thing that fixes #8
